packages/grid/test/sorting.common.js

```diff
@@ -144,24 +144,19 @@ describe('sorting', () => {
       expect(sorters.map((sorter) => sorter._order)).to.eql([2, 0, 1]);
     });
 
-    it('should not keep references to sorters when column is removed', () => {
+    it('should remove sorter reference when removing column', () => {
       grid.removeChild(columns[0]);
       flushGrid(grid);
       expect(grid._sorters).to.not.contain(sorters[0]);
     });
 
-    it('should update sorting when column is removed', () => {
+    it('should exclude removed column from sorting', () => {
       grid.removeChild(columns[2]);
       flushGrid(grid);
-
-      assertColumnCellOrder(columns[0], ['3', '1', '2']);
       assertColumnCellOrder(columns[1], ['1', '2', '3']);
     });
 
-    it('should update sort order when column removed and grid is not attached', () => {
```

Comment:
**note:** The sorters' order property becomes `null` for the time the grid is disconnected. It's a limitation of the new implementation, which can't reliably distinguish whether the column is hidden or the entire grid is hidden. I've assumed that it's not a big deal since the order property is restored to its original value when the grid is re-connected.